lib/CL/clCreateKernel.c

```diff
@@ -76,8 +78,10 @@ POname(clCreateKernel)(cl_program program,
       if (program->binaries[device_i] &&
           pocl_cache_device_cachedir_exists(program, device_i))
         {
+#ifdef OCS_AVAILABLE
           pocl_llvm_get_kernel_metadata(program, kernel, device_i,
                                         kernel_name, &errcode);
+#endif
```

Comment:
If you're creating a kernel form a program create with binary, it won't go in this section at all.
But as this function (pocl_llvm_get_kernel_metadata) doesn't exist in the offline compiler support version of POCL, I've just put macros around it to avoid modifying the code to much.